### PR TITLE
fix the habitat package promotion tests

### PR DESCRIPTION
**Pull request by robbkidd**
## Description

We had a few problems with the `habitat/test` pipeline. 

* The PowerShell scripts calling PowerShell scripts was apparently swallowing the errors being thrown from inner layers. Check the error level of those scripts and throw another error if need be.

* Test hosts haven't necessarily had "hab setup" run for \hab\bin to be on the system PATH. For the purposes on this test, we would like that to be the case. So, put Hab's bin directory at the start of the PATH and carry on.

* Move the minimum hab check/install to a reusable script.
  - `verify/habitat` needs hab >= 0.85 to both [build with environment variables that don't include the studio path](https://github.com/habitat-sh/habitat/pull/6850) and to [binlink the runtime
with stubs that include the package and system environment](https://github.com/habitat-sh/habitat/pull/6826).
  - `habitat/test` needs hab >= 0.85 to binlink the runtime with stubs that
include the package and system environment.

* Increase the timeout for `habitat/test`; the Windows hab tests take a long time.

## Types of changes

- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- ~I have updated the documentation accordingly.~
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by robbkidd** [comment]
`verify/build` and `habitat/test` both fail/pass as expected now. 👍

File: .expeditor/habitat-test.pipeline.yml [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
+      retry:
+        automatic:
+          limit: 1
+
 steps:
 
 - label: ":windows: Validate Habitat Builds of Chef Infra"
   commands:
+    - powershell -File "./scripts/ci/ensure-minimum-viable-hab.ps1"
+    - 'Write-Host "--- :hammer_and_wrench: Installing $$EXPEDITOR_PKG_IDENTS_CHEFINFRACLIENTX86_64WINDOWS"'
```
Comment (jeremiahsnapp) [inline .expeditor/habitat-test.pipeline.yml]: I think `$$EXPEDITOR_PKG_IDENTS_CHEFINFRACLIENTX86_64WINDOWS` isn't getting interpolated properly. One of the builds looked like it was blank.

Comment (robbkidd) [reply .expeditor/habitat-test.pipeline.yml]: Stupid double dollars! :rage1:

Comment (robbkidd) [reply .expeditor/habitat-test.pipeline.yml]: Stupid double dollars! :rage1: 

Updated with only one `$`.

**Pull request merged**